File: frontend-react/src/types/stats/YearlyFinancialSummary.ts
export interface YearlyFinancialSummary {
    monthlyExpenses: { [key: number]: number };
    monthlyIncomes: { [key: number]: number };
    monthlyExpenseTransactions: { [key: number]: number };
    monthlyIncomeTransactions: { [key: number]: number };
    minYear: number;
    maxYear: number;
}

File: src/main/java/org/balaur/financemanagement/response/stats/YearlyFinancialSummary.java
package org.balaur.financemanagement.response.stats;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Map;

@Data
@Builder
public class YearlyFinancialSummary {
    private Map<Integer, BigDecimal> monthlyExpenses;
    private Map<Integer, BigDecimal> monthlyIncomes;
    private Map<Integer, Long> monthlyExpenseTransactions;
    private Map<Integer, Long> monthlyIncomeTransactions;
    private Integer minYear;
    private Integer maxYear;
}
